starky/src/prover.rs

```diff
@@ -70,63 +69,113 @@ where
     challenger.observe_cap(&trace_cap);
 
     // Permutation arguments.
-    let permutation_zs_commitment_challenges = stark.uses_permutation_args().then(|| {
-        let permutation_challenge_sets = get_n_permutation_challenge_sets(
-            &mut challenger,
-            config.num_challenges,
-            stark.permutation_batch_size(),
-        );
-        let permutation_z_polys = compute_permutation_z_polys::<F, S, D>(
-            &stark,
-            config,
-            &trace_poly_values,
-            &permutation_challenge_sets,
-        );
+    let constraint_degree = stark.constraint_degree();
+    let lookups = stark.lookups();
+    let lookup_challenges = stark.uses_lookups().then(|| {
+        get_grand_product_challenge_set(&mut challenger, config.num_challenges)
+            .challenges
+            .iter()
+            .map(|ch| ch.beta)
+            .collect::<Vec<_>>()
+    });
+
+    let num_lookup_columns = lookups
+        .iter()
+        .map(|l| l.num_helper_columns(constraint_degree))
+        .sum();
 
-        let permutation_zs_commitment = timed!(
+    let auxiliary_polys_commitment = stark.uses_lookups().then(|| {
+        let lookup_helper_columns = timed!(timing, "compute lookup helper columns", {
+            let challenges = lookup_challenges.as_ref().expect("We do have challenges.");
+            let mut columns = Vec::with_capacity(num_lookup_columns);
```

Comment:
So the code on line 82 is only there to give the allocator a size hint? I wonder if that is worth the price of running though the `lookups` one extra time? Is it a small number perhaps? You could just "guesstimate" the size, and do `Vec::with_capacity(5000)` knowing it'll be mostly fine. If `columns` is going to survive this function call – seems like it will – then perhaps you can resize the vec before returning it?